sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -871,6 +886,23 @@ private[hive] object HiveClientImpl {
       hiveTable.setViewOriginalText(t)
       hiveTable.setViewExpandedText(t)
     }
+
+    table.bucketSpec match {
+      case Some(bucketSpec) =>
+        hiveTable.setNumBuckets(bucketSpec.numBuckets)
+        hiveTable.setBucketCols(bucketSpec.bucketColumnNames.toList.asJava)
```

Comment:
For data source table, which can be created by `CREATE TABLE src(...) USING parquet ...`, the bucketing information is in table properties, and hive will always read this table as a non-bucketed table.

After your PR, for bucketed data source tables written by Spark, Hive will read them as bucketed tables and cause problems, because the bucket hashing function is different.